cli/command/new/templates.go

```diff
@@ -1,74 +1,142 @@
+	"context"
+	"fmt"
+	"github.com/google/go-github/github"
+	"github.com/rs/zerolog/log"
+	"github.com/tj/survey"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+)
+
+var (
+	overWriteAll     = false
+	templateFileName = "saucetpl.tar.gz"
+)
+
+// GetReleaseArtifactURL provides template artifact url for a given repo
+func GetReleaseArtifactURL(org string, repo string) (string, error) {
+	ctx := context.Background()
+	ghClient := github.NewClient(nil)
+	release, _, err := ghClient.Repositories.GetLatestRelease(ctx, org, repo)
+	if err != nil {
+		return "", err
+	}
+
+	for _, asset := range release.Assets {
+		if *asset.Name == templateFileName {
+			return asset.GetBrowserDownloadURL(), nil
+		}
+	}
+	return "", fmt.Errorf("No %s found", templateFileName)
+}
+
+func createFolder(name string, mode int64) error {
+	err := os.MkdirAll(name, os.FileMode(mode))
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-var testTpl = map[string]SetupTemplate{
-	"testcafe": {
-		"example.test.js",
-		"import { Selector } from 'testcafe';\n" +
-			"fixture `Getting Started`\n" +
-			"	.page `http://devexpress.github.io/testcafe/example`\n\n" +
-			`const testName = 'testcafe test'
-test(testName, async t => {
-	await t
-		.typeText('#developer-name', 'devx')
-		.click('#submit-button')
-		.expect(Selector('#article-header').innerText).eql('Thank you, devx!');
-});
-	`},
-	"puppeteer": {
-		"example.test.js",
-		`describe('saucectl demo test', () => {
-	test('should verify title of the page', async () => {
-		const page = (await browser.pages())[0]
-		await page.goto('https://www.saucedemo.com/');
-		expect(await page.title()).toBe('Swag Labs');
-	});
-});
-`},
-	"playwright": {
-		"example.test.js",
-		`describe('saucectl demo test', () => {
-	test('should verify title of the page', async () => {
-		await page.goto('https://www.saucedemo.com/');
-		expect(await page.title()).toBe('Swag Labs');
-	});
-});
-`},
-	"cypress": {
-		"example.test.js",
-		`context('Actions', () => {
-		beforeEach(() => {
-			cy.visit('https://example.cypress.io/commands/actions')
-		})
-		it('.type() - type into a DOM element', () => {
-			// https://on.cypress.io/type
-			cy.get('.action-email')
-				.type('fake@email.com').should('have.value', 'fake@email.com')
-		})
-	})`},
+func confirmOverwriting(name string) bool {
+	if overWriteAll {
+		return true
+	}
+
+	var answer string
+	question := &survey.Select{
+		Message: fmt.Sprintf("Overwrite %s:", name),
+		Options: []string{"No", "Yes", "All"},
+		Default: "No",
+	}
+	err := survey.AskOne(question, &answer, nil)
+	if err != nil {
+		log.Err(err)
+		return false
+	}
+
+	overWriteAll = answer == "All"
+	return answer == "Yes" || answer == "All"
+}
+
+func extractFile(name string, mode int64, src io.Reader) error {
+	stat, err := os.Stat(name)
+	if err != nil && !os.IsNotExist(err) {
+		log.Err(err)
+		return err
+	}
+	if err == nil && stat.IsDir() {
+		return fmt.Errorf("%s exists and is a directory", name)
+	}
+
+	if err == nil {
```

Comment:
Imho it'd be better to refactor this entire method so that you didn't have to do these checks over and over again.
To the reader, it's not immediately clear that
```
	stat, err := os.Stat(name)
	if err != nil && !os.IsNotExist(err) {
```
is there to elicit an existance check that's primarily meant for the overwrite.

Reply:
I wrapped the existence check into a dedicated function to be more explicit about the purpose of this check.
It's more clear now.